src/numerics/include/metaphysicl/numberarray.h

```diff
@@ -130,6 +130,9 @@ class NumberArray
   typename entry_type<i>::type& insert()
     { return _data[i]; }
 
+  T & insert(std::size_t i)
+    { return _data[i]; }
```

Comment:
bounds-check assertion?

Reply:
I am being lazy there.  I'll add one.

Reply:
Done, and I went through the others while I was at it.

Even admitting this wasn't originally intended as a public project, it's painfully embarrassing to see that that header didn't have a single assertion yet.

Reply:
Haha. I think you can see you were mimicking the underlying data container as much as possible :smile: No protections on a C array!